content/en/docs/armory-agent/agent-vault.md

```diff
@@ -0,0 +1,163 @@
+        vault.hashicorp.com/agent-inject: "true"
+        vault.hashicorp.com/agent-inject-secret-kubesvc-local.yml: ""
+        vault.hashicorp.com/secret-volume-path-kubesvc-local.yml: '/opt/spinnaker/config'
+        vault.hashicorp.com/agent-inject-file-kubesvc-local.yml: 'kubesvc-local.yml'
+        vault.hashicorp.com/agent-inject-template-kubesvc-local.yml: |
+          kubernetes:
+            accounts:
+          {{- with secret "secret/kubernetes" -}}
+          {{ range $k, $v := .Data.data }}
+              - kubeconfigFile: 'encryptedFile:vault!e:secret!n:kubernetes!k:{{ $k }}'
+                name: {{ $k -}}
+          {{- else }}
+              []
+          {{ end -}}
+          {{- end }}
+          secrets:
+            vault:
+              enabled: true
+              authMethod: KUBERNETES
+              role:
+              path:
+              url:
+          clouddriver:
+            insecure: true
+    spec:
+      volumes:
+        - $patch: delete
+          name: volume-kubesvc-config
+        - $patch: delete
+          name: volume-kubesvc-kubeconfigs
+      containers:
+        - name: kubesvc
+          volumeMounts:
+            - $patch: delete
+              name: volume-kubesvc-config
+              mountPath: /opt/spinnaker/config
+            - $patch: delete
+              name: volume-kubesvc-kubeconfigs
+              mounthPath:
+              mountPath: /kubeconfigfiles
+{{</ prism >}}
+
+Considerations:
+ * Make sure to include the required [Vault Injector Annotations](https://www.vaultproject.io/docs/platform/k8s/injector/annotations) like [`vault.hashicorp.com/role` or `vault.hashicorp.com/agent-configmap`](https://www.vaultproject.io/docs/platform/k8s/injector/annotations#vault-hashicorp-com-role) correspond to your environment
+ * Be aware of the version of Vault's KV engine currently in your environment. This guide assumes you have the secret engine [KV version 2](https://www.vaultproject.io/docs/secrets/kv/kv-v2). For you KV version 1, will need to modify the template to use `{{ range $k, $v := .Data }}` instead. See [this link](https://github.com/hashicorp/consul-template/blob/master/docs/templating-language.md#versioned-read) for more information
+ * This template expects `secret/kubernets` to hold the kubeconfig file: Make sure to replace both line 16 and 18 in case that's not the case in your environment
+ * Make sure to include all other [Agent Options]({{< ref "agent-options/#options" >}}) that you might required in your environment
+
+After taking those considerations, save the template as `armory-agent-vault-patch.yaml`, and refer to it in your `kustomization.yaml`:
+
+
+{{< prism lang="yaml" line="10-11" >}}
+# ./kustomization.yaml
+# Pre-existing SpinnakerService resource (may be different)
+namespace: spinnaker
+resources:
+  - spinnakerservice.yaml
+bases:
+# Armory agent deployment
+  - https://armory.jfrog.io/artifactory/manifests/kubesvc/armory-agent-0.5.11-kustomize.tar.gz
```

Comment:
Do http references like this actually work? @th3morg was not able to make it work with recent versions of kustomize

Reply:
It works in kustomize 3.6.1 
I just noticed that kustomize 4.1.6 fails with `Error: accumulating resources: accumulation err='accumulating resources from 'https://armory.jfrog.io/artifactory/manifests/kubesvc/armory-agent-0.5.11-kustomize.tar.gz': yaml: control characters are not allowed': git cmd = '/usr/local/bin/git fetch --depth=1 origin HEAD': exit status 128`
kubectl kustomize does fail with `Error: rawResources failed to read`

Reply:
I'll update both

Reply:
updated